### PR TITLE
GraphQl: Normalize id based on the origin resource

### PR DESCRIPTION
**Pull request by soyuka**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | api-platform/api-platform#1045
| License       | MIT
| Doc PR        | na

When querying the following:

```
{
  language(id: "/api/languages/1") {
    id, foo
  }
}
```

where `Language` has an output it should get the following response:

```
{"data":{"language":{"id":"\api\/languages\1","foo":"bar"}}}
```

Where the `id` matches the origin resource IRI.

ping @phryneas

### CODE REVIEW

**Comment by phryneas** [comment]
@soyuka works fine for me for the `id` property - with another followup bug (sorry!): 
querying `_id` should return `1`, but returns `/api/languages/1`

File: src/GraphQl/Resolver/ResourceFieldResolver.php [hunk 1]
```diff
@@ -26,22 +28,29 @@
...
-            return $this->iriConverter->getIriFromItem(unserialize($source[ItemNormalizer::ITEM_KEY]));
+            $object = unserialize($source[ItemNormalizer::ITEM_KEY]);
+            if ($this->resourceClassResolver->isResourceClass($this->getObjectClass($object))) {
+                return $this->iriConverter->getIriFromItem($object);
+            }
         }
 
         if ('_id' === $info->fieldName && isset($source['id'])) {
-            $property = $source['id'];
+            $property = $source['_id'] ?? $source['id'];
```
Comment (alanpoulain) [inline src/GraphQl/Resolver/ResourceFieldResolver.php]: Add a comment why?

File: src/GraphQl/Serializer/ItemNormalizer.php [hunk 2]
```diff
@@ -63,6 +64,12 @@ public function normalize($object, $format = null, array $context = [])
             throw new UnexpectedValueException('Expected data to be an array');
         }
 
+        if (($context['origin_resource'] ?? false) && isset($data['id'])) {
```
Comment (alanpoulain) [inline src/GraphQl/Serializer/ItemNormalizer.php]: Missing comment too?

File: src/GraphQl/Type/SchemaBuilder.php [hunk 3]
```diff
@@ -401,7 +401,15 @@ private function convertType(Type $type, bool $input = false, string $mutationNa
...
     private function getResourceObjectType(?string $resourceClass, ResourceMetadata $resourceMetadata, bool $input = false, string $mutationName = null, int $depth = 0): GraphQLType
     {
-        $shortName = $resourceMetadata->getShortName();
+        $ioMetadata = $resourceMetadata->getAttribute($input ? 'input' : 'output');
+
+        if (null !== $ioMetadata && \array_key_exists('class', $ioMetadata) && null !== $ioMetadata['class']) {
+            $resourceClass = $ioMetadata['class'];
+            $shortName = $ioMetadata['name'];
+        } else {
+            $shortName = $resourceMetadata->getShortName();
```
Comment (alanpoulain) [inline src/GraphQl/Type/SchemaBuilder.php]: Before the if?

**Comment by soyuka** [comment]
Many thanks @teohhanhui for the review !

**Pull request merged**